crates/mzoon/src/wasm_opt.rs

```diff
@@ -97,23 +100,32 @@ async fn check_wasm_opt() {
 }
 
 #[throws]
-fn unpack_wasm_opt(tar_gz: Vec<u8>) {
+async fn unpack_wasm_opt(tar_gz: Vec<u8>) {
+    const LIBBINARYEN_PATH: &str = "frontend/binaryen/lib/libbinaryen.dylib";
+
     let tar = GzDecoder::new(tar_gz.as_slice());
     let mut archive = Archive::new(tar);
 
     for entry in archive.entries()? {
         let mut entry = entry?;
         let path = entry.path()?;
+        // Use file_stem() because windows executable has `.exe` extension
         let file_stem = path
             .file_stem()
             .ok_or(anyhow!("Entry without a file name"))?;
-        if file_stem != "wasm-opt" {
-            continue;
-        }
-        let mut destination = PathBuf::from("frontend");
-        destination.push(path.file_name().unwrap());
+
+        let destination = match file_stem.to_str() {
+            Some("wasm-opt") => {
+                PathBuf::from(WASM_OPT_PATH).with_file_name(path.file_name().unwrap())
+            }
+            Some("libbinaryen") => PathBuf::from(LIBBINARYEN_PATH),
```

Comment:
I've tried it but not successful. The method `.with_file_name()` yields an owned `PathBuf` which cannot be lived out of the match arm (this is just a repeat of compiler message...)

```
error[E0716]: temporary value dropped while borrowed
  --> src/main.rs:30:33
   |
29 |         let destination = match file_stem.to_str() {
   |             ----------- borrow later stored here
30 |             Some("wasm-opt") => Path::new(WASM_OPT_PATH).with_file_name(path.file_name().unwrap()).as_path(),
   |                                 ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^         - temporary value is freed at the end of this statement
   |                                 |
   |                                 creates a temporary which is freed while still in use
   |
   = note: consider using a `let` binding to create a longer lived value

```

I think it's possible to use `Path::new()` in the match arms and construct the actual path of the file name when unpacking like below. But I have no idea how to compare the trade off of these ways🤨

```rust
let destination = match file_stem.to_str() {
    Some("wasm-opt") => Path::new(WASM_OPT_PATH),
    Some("libbinaryen") => Path::new(LIBBINARYEN_PATH),
    _ => continue,
};

entry.unpack(destination.with_file_name(path.file_name().unwrap()))?;
```

Reply:
Yeah, I wasn't sure whether it fails this way or not. Then leave it as is / choose the simpler code :)

Reply:
FInally I use the `Path::new` way in https://github.com/MoonZoon/MoonZoon/pull/171#discussion_r1177724416 and added a comment on the `LIBBINARYEN_PATH` definition since I noticed the lib for Linux has different file extension. I would think constructing the actual file name on packing can be shared with both `wasm-opt` and `libbinaryen` that being a little simpler.